straxen/plugins/afterpulse_processing.py

```diff
@@ -0,0 +1,345 @@
+            if not buffer[offset]['time'] == 0:
+                offset += 1
+            prev_record_i = h['record_i']
+            record_data = records[prev_record_i]['data']
+            baseline_fpart = records[prev_record_i]['baseline'] % 1
+        
+        res = buffer[offset]
+
+            
+        if h['left'] < LED_window_left:
+            # if hit is before LED window: discard
+            continue
+            
+        if (h['left'] >= LED_window_left) & (h['left'] < LED_window_right):
+            # hit is in LED window
+            if not is_LED:
+                # this is the first hit in the LED window
+                res['time'] = h['time']
+                res['dt'] = h['dt']
+                res['channel'] = h['channel']
+                res['left'] = h['left']
+                res['right'] = h['right']
+                res['record_i'] = h['record_i']
+                res['threshold'] = h['threshold']
+                res['height'] = h['height']
+                
+                res['left_integration'] =  h['left'] - hit_left_extension
+                res['right_integration'] = h['right'] + hit_right_extension
+
+                res['length'] = res['right_integration'] - res['left_integration']
+                
+                # need to add baseline_fpart to area
+                hit_data = record_data[res['left_integration']:res['right_integration']]
+                res['area'] = hit_data.sum() + res['length'] * baseline_fpart
+                
+                res['sample_10pc_area'] = res['left_integration'] + get_sample_area_quantile(hit_data, 0.1, baseline_fpart)
+                res['sample_50pc_area'] = res['left_integration'] + get_sample_area_quantile(hit_data, 0.5, baseline_fpart)
+                
+                res['max'] = res['left_integration'] + hit_data.argmax()
+                
+                # set the LED time in the current WF
+                t_LED = res['sample_10pc_area']
+                
+                is_LED = True
+                
+                continue
+            
+            # more hits in LED window: extend the first (merging all hits in the LED window)
+            res['right'] = h['right']
+            res['right_integration'] = h['right'] + hit_right_extension
+
+            res['length'] = res['right_integration'] - res['left_integration']
+            res['height'] = max(res['height'], h['height'])
+            
+            hit_data = record_data[res['left_integration']:res['right_integration']]
+            res['area'] = hit_data.sum() + res['length'] * baseline_fpart
+
+            res['sample_10pc_area'] = res['left_integration'] + get_sample_area_quantile(hit_data, 0.1, baseline_fpart)
+            res['sample_50pc_area'] = res['left_integration'] + get_sample_area_quantile(hit_data, 0.5, baseline_fpart)
+
+            res['max'] = res['left_integration'] + hit_data.argmax()
+
+            t_LED = res['sample_10pc_area']
```

Comment:
I am not sure if this is possible. But to me it looks like that this fraction of the code is required several times. Maybe it is worth to be put in a small extra function.